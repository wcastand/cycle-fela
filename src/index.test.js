import { run } from '@cycle/run'
import { createRenderer } from 'fela'
import { div, span, hr, h } from '@cycle/dom'
import xs from 'xstream'

import { makeFelaDomDriver, createComponent } from './index'

const styleNode = document.createElement('style')
document.head.appendChild(styleNode)

function createBase(selector) {
  const el = document.createElement('div')
  el.id = selector
  document.body.appendChild(el)
  return el
}
function createDriver(selector) {
  return makeFelaDomDriver(createBase(selector), { customStyleNode: styleNode })
}

describe('Cycle app', () => {
  it('should render corectly', done => {
    const RedLabel = createComponent(() => ({ color: 'red' }), 'label')
    const Container = createComponent(({ mobile = true }) => ({
      display: 'flex',
      flex: mobile ? '1' : '2',
    }))
    function main(sources) {
      const vdom$ = xs.of(
        Container({ mobile: false }, [RedLabel('test'), span('Name:'), hr(), span(`Hello`)]),
      )
      return { DOM: vdom$ }
    }

    run(main, { DOM: createDriver('app') })

    setTimeout(() => {
      const style = document.querySelector('style')
      const dom = document.querySelector('#app')

      expect(style).toMatchSnapshot()
      expect(dom).toMatchSnapshot()
      done()
    }, 150)
  })

  it('should have a dom and a style', done => {
    function main(sources) {
      const vdom$ = xs.of(
        div([
          h('div', { component: () => ({ backgroundColor: 'red' }) }, 'test'),
          span('Name:'),
          hr(),
          span(`Hello`),
        ]),
      )
      return { DOM: vdom$ }
    }

    run(main, { DOM: createDriver('app-test') })

    setTimeout(() => {
      const style = document.querySelector('style')
      const dom = document.querySelector('#app-test')

      expect(style).toMatchSnapshot()
      expect(dom).toMatchSnapshot()
      done()
    }, 150)
  })
})
