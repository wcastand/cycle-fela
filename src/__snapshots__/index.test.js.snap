// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cycle app should have a dom and a style 1`] = `
<style
  data-fela-stylesheet=""
>
  .a{background-color:red}
</style>
`;

exports[`Cycle app should have a dom and a style 2`] = `
<div
  id="app-test"
>
  <div>
    <div
      class="a"
    >
      test
    </div>
    <span>
      Name:
    </span>
    <hr />
    <span>
      Hello
    </span>
  </div>
</div>
`;

exports[`Cycle app should pass a theme prop to createComponent 1`] = `
<style
  data-fela-stylesheet=""
>
  .a{color:red}.b{display:flex}.c{flex:2}
</style>
`;

exports[`Cycle app should pass a theme prop to createComponent 2`] = `
<div
  id="app"
>
  <div
    class="b c"
  >
    <label
      class="a"
    >
      test
    </label>
    <span>
      Name:
    </span>
    <hr />
    <span>
      Hello
    </span>
  </div>
</div>
`;

exports[`Cycle app should render corectly 1`] = `
<style
  data-fela-stylesheet=""
>
  .a{color:red}.b{display:flex}.c{flex:2}
</style>
`;

exports[`Cycle app should render corectly 2`] = `
<div
  id="app"
>
  <div
    class="b c"
  >
    <label
      class="a"
    >
      test
    </label>
    <span>
      Name:
    </span>
    <hr />
    <span>
      Hello
    </span>
  </div>
</div>
`;
